### ISSUE TITLE
Incompatibility issue with pyensight (0.6.1) and debugpy (1.8.0)

### ISSUE THREAD

**Issue by ansRMorin** [ansys/pyensight#326]
### 🔍 Before submitting the issue

- [X] I have searched among the existing issues
- [X] I am using a Python virtual environment

### 🐞 Description of the bug

We are using pysensight in a SAF-based Solution Application which has debugpy as a dependency and there seems to be an incompatibility between pyensight and debugpy. Installing both dependencies in a virtual environment and running a print command results in a timeout exception.
![image](https://github.com/ansys/pyensight/assets/106841265/1bbc1d6d-a132-4964-98e0-d13bd1888950)

### 📝 Steps to reproduce

To reproduce the issue:

1. python -m venv .venv_pyensight (tested with python 3.8.10)
2. cd .venv_pyensight
3. .\Scripts\activate
4. pip install --upgrade pip
5. pip install debugpy==1.8.0
6. python -m debugpy --listen 5768 -c "print('hello')" (should just print hello to console)
7. pip install ansys-pyensight-core==0.6.1
8. python -m debugpy --listen 5768 -c "print('hello')" (the timeout exception appears)

This is the debug log file:

```
I+00000.015: Windows-10-10.0.22621-SP0 AMD64
             CPython 3.8.10 (64-bit)
             debugpy 1.8.0

I+00000.047: Initial environment:
             
             System paths:
                 sys.executable: C:\Users\aciurana\git\.venv_pyensight\Scripts\python.exe
                 sys.prefix: C:\Users\aciurana\git\.venv_pyensight
                 sys.base_prefix: C:\Users\aciurana\.pyenv\pyenv-win\versions\3.8.10
                 sys.real_prefix: <missing>
                 site.getsitepackages(): C:\Users\aciurana\git\.venv_pyensight
                                         C:\Users\aciurana\git\.venv_pyensight\lib\site-packages(C:\Users\aciurana\git\.venv_pyensight\Lib\site-packages)
                 site.getusersitepackages(): C:\Users\aciurana\AppData\Roaming\Python\Python38\site-packages
                 sys.path (site-packages): C:\Users\aciurana\git\.venv_pyensight\lib\site-packages(C:\Users\aciurana\git\.venv_pyensight\Lib\site-packages)
                 sysconfig.get_path('stdlib'): C:\Users\aciurana\.pyenv\pyenv-win\versions\3.8.10\Lib
                 sysconfig.get_path('platstdlib'): C:\Users\aciurana\git\.venv_pyensight\Lib
                 sysconfig.get_path('purelib'): C:\Users\aciurana\git\.venv_pyensight\Lib\site-packages
                 sysconfig.get_path('platlib'): C:\Users\aciurana\git\.venv_pyensight\Lib\site-packages
                 sysconfig.get_path('include'): C:\Users\aciurana\.pyenv\pyenv-win\versions\3.8.10\Include(C:\Users\aciurana\.pyenv\pyenv-win\versions\3.8.10\include)
                 sysconfig.get_path('scripts'): C:\Users\aciurana\git\.venv_pyensight\Scripts
                 sysconfig.get_path('data'): C:\Users\aciurana\git\.venv_pyensight
                 os.__file__: C:\Users\aciurana\.pyenv\pyenv-win\versions\3.8.10\lib\os.py(C:\Users\aciurana\.pyenv\pyenv-win\versions\3.8.10\Lib\os.py)
                 threading.__file__: C:\Users\aciurana\.pyenv\pyenv-win\versions\3.8.10\lib\threading.py(C:\Users\aciurana\.pyenv\pyenv-win\versions\3.8.10\Lib\threading.py)
                 debugpy.__file__: C:\Users\aciurana\git\.venv_pyensight\lib\site-packages\debugpy\__init__.py(C:\Users\aciurana\git\.venv_pyensight\Lib\site-packages\debugpy\__init__.py)
             
             Installed packages:
                 ansys-api-pyensight==0.3.2
                 ansys-pyensight-core==0.6.1
                 certifi==2023.7.22
                 charset-normalizer==3.3.0
                 debugpy==1.8.0
                 grpcio==1.59.0
                 idna==3.4
                 importlib-metadata==6.8.0
                 numpy==1.24.4
                 Pillow==10.0.1
                 pip==23.2.1
                 protobuf==4.24.4
                 pyansys-docker==5.0.4
                 pywin32==306
                 requests==2.31.0
                 setuptools==56.0.0
                 typing==3.7.4.3
                 typing_extensions==4.8.0
                 urllib3==1.26.17
                 websocket-client==1.6.4
                 zipp==3.17.0

I+00000.047: sys.argv before parsing: ['C:\\Users\\aciurana\\git\\.venv_pyensight\\lib\\site-packages\\debugpy\\__main__.py', '--listen', '5768', '--log-to', './', '-c', "print('hello')"]
                      after parsing:  ['C:\\Users\\aciurana\\git\\.venv_pyensight\\lib\\site-packages\\debugpy\\__main__.py']

D+00000.047: sys.argv after patching: ['-c']

D+00000.047: configure({'qt': 'none', 'subProcess': True}, {})

D+00000.047: listen(('127.0.0.1', 5768), **{})

I+00000.047: Initial debug configuration: {
                 "qt": "none",
                 "subProcess": true,
                 "python": "C:\\Users\\aciurana\\git\\.venv_pyensight\\Scripts\\python.exe",
                 "pythonEnv": {}
             }

I+00000.047: Waiting for adapter endpoints on 127.0.0.1:52713...

I+00000.047: debugpy.listen() spawning adapter: [
                 "C:\\Users\\aciurana\\git\\.venv_pyensight\\Scripts\\python.exe",
                 "C:\\Users\\aciurana\\git\\.venv_pyensight\\lib\\site-packages\\debugpy\\adapter",
                 "--for-server",
                 "52713",
                 "--host",
                 "127.0.0.1",
                 "--port",
                 "5768",
                 "--server-access-token",
                 "fa18704fae5661444e6623d3598c168a2e2bd917570c3e8a954ed18d762a8715",
                 "--log-dir",
                 "./"
             ]

I+00010.062: Timed out waiting for adapter to connect:
             
             Traceback (most recent call last):
               File "C:\Users\aciurana\git\.venv_pyensight\lib\site-packages\debugpy\server\api.py", line 237, in listen
                 sock, _ = endpoints_listener.accept()
               File "C:\Users\aciurana\.pyenv\pyenv-win\versions\3.8.10\lib\socket.py", line 292, in accept
                 fd, addr = self._accept()
             socket.timeout: timed out
             
             Stack where logged:
               File "C:\Users\aciurana\.pyenv\pyenv-win\versions\3.8.10\lib\runpy.py", line 194, in _run_module_as_main
                 return _run_code(code, main_globals, None,
               File "C:\Users\aciurana\.pyenv\pyenv-win\versions\3.8.10\lib\runpy.py", line 87, in _run_code
                 exec(code, run_globals)
               File "C:\Users\aciurana\git\.venv_pyensight\lib\site-packages\debugpy\__main__.py", line 39, in <module>
                 cli.main()
               File "C:\Users\aciurana\git\.venv_pyensight\lib\site-packages\debugpy\server\cli.py", line 430, in main
                 run()
               File "C:\Users\aciurana\git\.venv_pyensight\lib\site-packages\debugpy\server\cli.py", line 325, in run_code
                 start_debugging(str("-c"))
               File "C:\Users\aciurana\git\.venv_pyensight\lib\site-packages\debugpy\server\cli.py", line 256, in start_debugging
                 debugpy.listen(options.address)
               File "C:\Users\aciurana\git\.venv_pyensight\lib\site-packages\debugpy\public_api.py", line 31, in wrapper
                 return wrapped(*args, **kwargs)
               File "C:\Users\aciurana\git\.venv_pyensight\lib\site-packages\debugpy\server\api.py", line 141, in debug
                 return func(address, settrace_kwargs, **kwargs)
               File "C:\Users\aciurana\git\.venv_pyensight\lib\site-packages\debugpy\server\api.py", line 248, in listen
                 log.swallow_exception(
               File "C:\Users\aciurana\git\.venv_pyensight\lib\site-packages\debugpy\common\log.py", line 215, in swallow_exception
                 _exception(format_string, *args, **kwargs)
             

I+00010.062: listen() failed:
             
             Traceback (most recent call last):
               File "C:\Users\aciurana\git\.venv_pyensight\lib\site-packages\debugpy\server\api.py", line 237, in listen
                 sock, _ = endpoints_listener.accept()
               File "C:\Users\aciurana\.pyenv\pyenv-win\versions\3.8.10\lib\socket.py", line 292, in accept
                 fd, addr = self._accept()
             socket.timeout: timed out
             
             During handling of the above exception, another exception occurred:
             
             Traceback (most recent call last):
               File "C:\Users\aciurana\git\.venv_pyensight\lib\site-packages\debugpy\server\api.py", line 141, in debug
                 return func(address, settrace_kwargs, **kwargs)
               File "C:\Users\aciurana\git\.venv_pyensight\lib\site-packages\debugpy\server\api.py", line 251, in listen
                 raise RuntimeError("timed out waiting for adapter to connect")
             RuntimeError: timed out waiting for adapter to connect
             
             Stack where logged:
               File "C:\Users\aciurana\.pyenv\pyenv-win\versions\3.8.10\lib\runpy.py", line 194, in _run_module_as_main
                 return _run_code(code, main_globals, None,
               File "C:\Users\aciurana\.pyenv\pyenv-win\versions\3.8.10\lib\runpy.py", line 87, in _run_code
                 exec(code, run_globals)
               File "C:\Users\aciurana\git\.venv_pyensight\lib\site-packages\debugpy\__main__.py", line 39, in <module>
                 cli.main()
               File "C:\Users\aciurana\git\.venv_pyensight\lib\site-packages\debugpy\server\cli.py", line 430, in main
                 run()
               File "C:\Users\aciurana\git\.venv_pyensight\lib\site-packages\debugpy\server\cli.py", line 325, in run_code
                 start_debugging(str("-c"))
               File "C:\Users\aciurana\git\.venv_pyensight\lib\site-packages\debugpy\server\cli.py", line 256, in start_debugging
                 debugpy.listen(options.address)
               File "C:\Users\aciurana\git\.venv_pyensight\lib\site-packages\debugpy\public_api.py", line 31, in wrapper
                 return wrapped(*args, **kwargs)
               File "C:\Users\aciurana\git\.venv_pyensight\lib\site-packages\debugpy\server\api.py", line 143, in debug
                 log.reraise_exception("{0}() failed:", func.__name__, level="info")
               File "C:\Users\aciurana\git\.venv_pyensight\lib\site-packages\debugpy\common\log.py", line 222, in reraise_exception
                 _exception(format_string, *args, **kwargs)
             

I+00010.078: Not logging to "<stderr>" anymore.
```

### 💻 Which operating system are you using?

Windows

### 🐍 Which Python version are you using?

3.8
It's also good to mention that this was tested with python versions 3.8.10, 3.9.13, 3.10.10, and 3.11.2 and it only works with 3.9.13.

### 📦 Installed packages

```shell
ansys-api-pyensight==0.3.2
ansys-pyensight-core==0.6.1
certifi==2023.7.22
charset-normalizer==3.3.0  
debugpy==1.8.0
grpcio==1.59.0
idna==3.4
importlib-metadata==6.8.0  
numpy==1.24.4
Pillow==10.0.1
pipdeptree==2.13.0
protobuf==4.24.4
pyansys-docker==5.0.4      
pywin32==306
requests==2.31.0
typing==3.7.4.3
typing_extensions==4.8.0   
urllib3==1.26.17
websocket-client==1.6.4
zipp==3.17.0
```

**Comment by randallfrank** [comment]
Yes, this is a known issue with debugpy.  It is a side effect of dependencies on the typing modules.  I don't remember which one, but either 'typing' or 'typing_extensions' moved from being an external module to being a built-in module in Python 3.9.  This is why it works with Python 3.9, but has issues with 3.8.  

@ansRMorin Could you please running:  

python -m pip uninstall typing

after installing pyensight but before trying to use pydebug?

We do this in the EnSight cpython interpreter to work around the problem.  I'm not sure that will work with 3.8,, but it should work with 3.9 and beyond.

**Comment by ansRMorin** [comment]
Thank you @randallfrank. I tested this with python 3.8.10 (pip uninstall typing) and python 3.10.10 (pip uninstall typing_extensions) and this workaround seems to be working for both versions.

**Comment by randallfrank** [comment]
I'm not entirely sure what we should do here.  This is a recently introduced issue with the pydebug module and older versions of the pydebug module do not have the problem.  We are kind of hoping that the pydebug devs will fix the problem, but so far that has not happened.  We could generate different wheels for different versions of Python, but we'd really like to avoid that if we can.  @mariostieriansys will the more complex dependency be a workable solution?

**Comment by mariostieriansys** [comment]
While in theory it would be possible to generate different wheels for different python versions, this would complicate quite a bit the workflow.

Since it is open source, I would rather open an issue on the debugpy repository.

Mario


Mario Ostieri  /  R&D Engineer II

 [cid:2f01b00f-6fba-48bc-925b-ea8f96bc3ea3]

97 Jubilee Avenue  /  Milton Park, Abingdon, OX14 4RW

Mobile (+44) 07735411454  /  office (+44) 01235 432257

***@***.***  /   www.ansys.com<http://www.ansys.com/>

--------------------------------------------------------------

The information transmitted is intended only for the person or entity to which it is addressed and may contain confidential and/or privileged material. Any review, retransmission, dissemination or other use of, or taking of any action in reliance upon this information by persons or entities other than the intended recipient is prohibited. If you received this in error, please contact the sender and delete the material from any computer.

ANSYS UK Limited Company Registration No: 3917515 VAT No: GB 471 0875 44 is a member of the ANSYS, Inc. group and is registered in England and Wales. Registered Office: 97 Jubilee Avenue, Milton Park, Abingdon, Oxfordshire OX14 4RW

________________________________

**Comment by mariostieriansys** [comment]
Given the nature of the issue, that can be resolved uninstalling typing, and given the fact that the support for 3.8 is being dropped, I am closing this issue

**Comment by KratikaJn** [comment]
We upgraded to pyensight v0.7.0 and glow 0.4.0 and seeing the same issues with debugpy. Since we are using typing and typing_extensions in a stage of the Volitare project and cannot remove the same, is there another workaround to resolve this issue?

**Comment by mariostieriansys** [comment]
As of release v0.7.0 we have removed the support for Python 3.8.
This means that typing is pre-installed in the python distribution and we don't need to list it as a dependency.
Also, I have removed the dependency on typing_extensions.

I have published a PR, and once merged I will create a 0.7.1 release that should solve your issue.

Mario


Mario Ostieri  /  R&D Engineer II

 [cid:ddf68cf7-97fe-4ff6-bd6e-4d6675467656]

97 Jubilee Avenue  /  Milton Park, Abingdon, OX14 4RW

Mobile (+44) 07735411454  /  office (+44) 01235 432257

***@***.***  /   www.ansys.com<http://www.ansys.com/>

--------------------------------------------------------------

The information transmitted is intended only for the person or entity to which it is addressed and may contain confidential and/or privileged material. Any review, retransmission, dissemination or other use of, or taking of any action in reliance upon this information by persons or entities other than the intended recipient is prohibited. If you received this in error, please contact the sender and delete the material from any computer.

ANSYS UK Limited Company Registration No: 3917515 VAT No: GB 471 0875 44 is a member of the ANSYS, Inc. group and is registered in England and Wales. Registered Office: 97 Jubilee Avenue, Milton Park, Abingdon, Oxfordshire OX14 4RW

________________________________

**Comment by mariostieriansys** [comment]
Hi Kratika,
I can confirm that PyEnSight 0.7.1 has been released, which removes the dependencies from typing and typing_extensions.
As noted earlier, PyEnSight dropped the support for Python 3.8 so you'll need to use at least Python 3.9, but apart from that your debugpy dependency now should be fine.

Thanks,
Mario



Mario Ostieri  /  R&D Engineer II

 [cid:efd23713-7771-47c5-a22f-8277aed23d75]

3 St Paul's Place, 129 Norfolk Street  /  Sheffield, England, S1 2JE

Mobile (+44) 07735411454  /  office (+44) 01235 432257

***@***.***  /   www.ansys.com<http://www.ansys.com/>

--------------------------------------------------------------

The information transmitted is intended only for the person or entity to which it is addressed and may contain confidential and/or privileged material. Any review, retransmission, dissemination or other use of, or taking of any action in reliance upon this information by persons or entities other than the intended recipient is prohibited. If you received this in error, please contact the sender and delete the material from any computer.

ANSYS UK Limited Company Registration No: 3917515 VAT No: GB 471 0875 44 is a member of the ANSYS, Inc. group and is registered in England and Wales. Registered Office: 3 St Paul's Place, 129 Norfolk Street, Sheffield, England, S1 2JE

________________________________
